src/main/java/com/github/javabdd/BDDFactoryIntImpl.java

```diff
@@ -86,6 +86,16 @@ public abstract class BDDFactoryIntImpl extends BDDFactory {
 
     protected abstract /* bdd */int veccompose_impl(/* bdd */int v, BDDPairing p);
 
+    protected abstract /* bdd */int relnext_impl(/* bdd */int states, /* bdd */int relation, /* bdd */int vars);
+
+    protected abstract /* bdd */int relnextRestricted_impl(/* bdd */int states, /* bdd */int relation,
+            /* bdd */int restriction, /* bdd */int vars);
+
+    protected abstract /* bdd */int relprev_impl(/* bdd */int relation, /* bdd */int states, /* bdd */int vars);
+
+    protected abstract /* bdd */int relprevRestricted_impl(/* bdd */int relation, /* bdd */int states,
```

Comment:
Is there a reason the `states` and `relation` arguments are swapped between `relprev*` and `relnext*`? Not really a problem, just curious.

Reply:
This is taken over from Sylvan like this. In the `BDD` class `relnext` and `relprev` have the same signature. So users of JavaBDD would not really notice.

Reply:
Indeed, I saw it was internal. Thanks for the clarification. Let's keep it as is.